alpha/lib/model/LiveEntry.php

```diff
@@ -577,23 +575,29 @@ private static function getCacheType()
 	{
 		return kCacheManager::CACHE_TYPE_LIVE_MEDIA_SERVER . '_' . kDataCenterMgr::getCurrentDcId();
 	}
-	
-	private function isCacheValid(kLiveMediaServer $kMediaServer)
+
+	/**
+	 * @param LiveEntryServerNode $liveEntryServerNode
+	 * @return bool|mixed
+	 * @throws Exception
+	 */
+	private function isCacheValid($liveEntryServerNode)
 	{
 		$cacheType = self::getCacheType();
 		$cacheStore = kCacheManager::getSingleLayerCache($cacheType);
 		if(! $cacheStore)
 		{
 			KalturaLog::warning("Cache store [$cacheType] not found");
-			$lastUpdate = time() - $kMediaServer->getTime();
+			$lastUpdate = time() - $liveEntryServerNode->getUpdatedAt(null);
```

Comment:
Note that on the kMediaServer getTime always returned the same value ... the time on kMediaServer was set in the __construct therefore never changed.

I agree that this change fixes a bug here that we never noticed but we need to make sure this wont trigger any hidden issues that we never noticed.

Please verify.